src/light_push.rs

```diff
@@ -24,37 +21,31 @@ pub struct Codec {}
 #[async_trait]
 impl request_response::Codec for Codec {
     type Protocol = StreamProtocol;
-    type Request = messages::PushRequest;
-    type Response = messages::PushResponse;
+    type Request = messages::PushRpc;
+    type Response = messages::PushRpc;
 
-    async fn read_request<T>(
-        &mut self,
-        _: &Self::Protocol,
-        io: &mut T,
-    ) -> io::Result<messages::PushRequest>
+    async fn read_request<T>(&mut self, _: &Self::Protocol, io: &mut T) -> io::Result<Self::Request>
     where
         T: AsyncRead + Unpin + Send,
     {
-        let mut vec = Vec::new();
-
-        io.take(REQUEST_SIZE_MAXIMUM).read_to_end(&mut vec).await?;
-
-        let request = messages::PushRequest::decode(&vec[..])?;
+        let mut vec = Vec::with_capacity(MAX_LIGHTPUSH_RPC_SIZE as usize);
+        io.read_to_end(&mut vec).await?;
+        let request = Self::Request::decode_length_delimited(&vec[..])?;
```

Comment:
was using `decode_length_delimited` needed to get it to work, or is it just a better way of doing it?